Support for cacheman options (fromDate cache property (in ticks)

fiolkaf opened the pull request.

fiolkaf committed:
from date support

nandanrao commented on lib/index.js:
```diff
@@ -31,6 +42,11 @@ var set = function (key, data, ttl, fn) {
 
+  if (fromDate) {
+    setParams.Metadata = {
+      FromDate: fromDate
```
Either pass in time and change the name of this to something more generic, OR take away passed in time and rely on AWS's modifiedDate.

nandanrao commented on lib/index.js:
```diff
@@ -58,7 +81,13 @@ var get = function (key, fn) {
     }
     if (data.Expires) {
```
We should ignore the expiry IF we're given a fromDate. --> ??

nandanrao commented on lib/index.js:
```diff
@@ -58,7 +81,13 @@ var get = function (key, fn) {
+    }
+    var fromDate = options.fromDate;
+    if (data.Metadata && data.Metadata.FromDate) {
+      if (data.Metadata.FromDate < options.fromDate) {
```
change to AWS modified date

nandanrao committed:
Fixed fromDate and cleaned up tests.

- fromDate now uses the LastModified property of S3.
- Fixed the tests to actually test business logic, not the mock.

nandanrao committed:
fixed function docs -- optional ttl
